code/game/machinery/defib_mount.dm

```diff
@@ -0,0 +1,156 @@
+//Holds defibs and recharges them from the powernet
+//You can activate the mount with an empty hand to grab the paddles
+//Not being adjacent will cause the paddles to snap back
+/obj/machinery/defibrillator_mount
+	name = "defibrillator mount"
+	desc = "Holds and recharges defibrillators. You can grab the paddles if one is mounted."
+	icon = 'icons/obj/machines/defib_mount.dmi'
+	icon_state = "defibrillator_mount"
+	density = FALSE
+	use_power = IDLE_POWER_USE
+	idle_power_usage = 1
+	power_channel = EQUIP
+	req_one_access = list(access_medical, access_heads) //used to control clamps
+	var/obj/item/defibrillator/defib //this mount's defibrillator
+	var/clamps_locked = FALSE //if true, and a defib is loaded, it can't be removed without unlocking the clamps
+
+/obj/machinery/defibrillator_mount/New(location, direction, building = 0)
+	. = ..()
+
+	if(location)
+		loc = location
+
+	if(direction)
+		dir = direction
+
+	if(building)
+		pixel_x = (dir & 3)? 0 : (dir == 4 ? -30 : 30)
+		pixel_y = (dir & 3)? (dir == 1 ? -30 : 30) : 0
+
+
+/obj/machinery/defibrillator_mount/loaded/New() //loaded subtype for mapping use
+	. = ..()
+	defib = new/obj/item/defibrillator/loaded(src)
+	update_icon()
+
+/obj/machinery/defibrillator_mount/Destroy()
+	if(defib)
+		QDEL_NULL(defib)
+	. = ..()
+
+/obj/machinery/defibrillator_mount/examine(mob/user)
+	..()
+	if(defib)
+		to_chat(user, "<span class='notice'>There is a defib unit hooked up. Alt-click to remove it.<span>")
+		if(security_level >= SEC_LEVEL_RED)
+			to_chat(user, "<span class='notice'>Due to a security situation, its locking clamps can be toggled by swiping any ID.</span>")
+		else
+			to_chat(user, "<span class='notice'>Its locking clamps can be [clamps_locked ? "dis" : ""]engaged by swiping an ID with access.</span>")
+	else
+		to_chat(user, "<span class='notice'>There are a pair of <b>bolts</b> in the defib unit housing securing the [src] to the wall.<span>")
+
+/obj/machinery/defibrillator_mount/process()
+	if(defib && defib.bcell && defib.bcell.charge < defib.bcell.maxcharge && is_operational())
+		use_power(200)
+		defib.bcell.give(180) //90% efficiency, slightly better than the cell charger's 87.5%
+		update_icon()
+
+/obj/machinery/defibrillator_mount/update_icon()
+	cut_overlays()
```

Comment:
I'm not sure how well the `cut_overlays` and `add_overlays` system works with objects. I know it was implemented for mobs.

Just be doubly sure this doesn't cause any oddities or runtimes.

Reply:
This is how syringes and similar reagent containers handle it AFAIK.

Reply:
they use `overlays.Cut()` and `overlays += "thing"`

`cut_overlays` is a TG proc with their icon SS. as is `add_overlay`

Reply:
I think I messed with `cut_overlays()` in one of my little side projects and it didn't work until someone changed it to `overlays.cut()`. I think Fox is correct

Reply:
These procs are implemented at the atom level and work as expected in this case.